lib/guardian.rb

```diff
@@ -529,13 +529,14 @@ def can_see_site_contact_details?
   end
 
   def auth_token
-    cookie_string = request&.cookies[Auth::DefaultCurrentUserProvider::TOKEN_COOKIE].presence
+    cookie_name = Auth::DefaultCurrentUserProvider::TOKEN_COOKIE
+    cookie_string = request&.cookies[cookie_name].presence
     if cookie_string
-      cookie = DiscourseAuthCookie.parse(cookie_string)
-      UserAuthToken.hash_token(cookie.token)
+      req = ActionDispatch::Request.new(request.env)
+      cookie = req.cookie_jar.encrypted[cookie_name]
+      return if !cookie
+      UserAuthToken.hash_token(cookie[:token])
     end
```

Comment:
In the guardian, `request` is already an `ActionDispatch::Request`, so I think we can just do
```suggestion
    cookie_name = Auth::DefaultCurrentUserProvider::TOKEN_COOKIE
    cookie = request&.cookie_jar.encrypted[cookie_name].presence
    return if !cookie
    UserAuthToken.hash_token(cookie[:token])
```